lib/v2/ymgal/article.js

```diff
@@ -0,0 +1,47 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+
+const host = 'https://www.ymgal.games';
+
+const types = {
+    news: '?type=NEWS&page=1',
+    column: '?type=COLUMN&page=1',
+};
+
+module.exports = async (ctx) => {
+    const type = ctx.params.type || 'news';
+    const link = `${host}/co/topic/list` + types[type];
+
+    const response = await got(link);
+    const data = response.data.data;
+
+    const items = await Promise.all(
+        data.map(async (item) => {
+            const itemUrl = host + '/co/article/' + item.topicId;
+            const description = await ctx.cache.tryGet(item.topicId, async () => {
+                const result = await got.get(itemUrl);
+                const $ = cheerio.load(result.data);
+                return $('article').html().trim();
+            });
+            const single = {
+                title: item.title,
+                link: itemUrl,
+                pubDate: new Date(item.publishTime).toUTCString(),
+                description,
+            };
+            return Promise.resolve(single);
```

Comment:
```suggestion
        data.map((item) => {
            const itemUrl = host + '/co/article/' + item.topicId;
            return ctx.cache.tryGet(itemUrl, async () => {
                const result = await got(itemUrl);
                const $ = cheerio.load(result.data);
                const description = $('article').html().trim();
                return {
                    title: item.title,
                    link: itemUrl,
                    pubDate: timezone(parseDate(item.publishTime), 8),
                    description,
                };
            });
```
- You can use `itemUrl` as the cache key instead of `topicId` since it has a lower chance of collision and provides more details for those managing Redis.
- You can cache the title, link and pubDate as an object as well.
- The `publishTime` provided from the API may need an offset after comparing with the article